test/kdump_test.rb

```diff
@@ -159,4 +167,290 @@
+      end
+
+      it "does not find several ranges" do
+        expect(several_ranges).to eq false
+      end
+
+      it "does not schedule writing on the param" do
+        expect(crashkernel_needed).to eq false
+      end
+    end
+
+    context "when the param is set to true" do
+      let(:kernel_param) { :present }
+
+      it "reports presence of the param" do
+        expect(is_crashkernel_set).to eq true
+      end
+
+      it "schedules the writing on the param" do
+        expect(crashkernel_needed).to eq true
+      end
+
+      it "does not find several ranges" do
+        expect(several_ranges).to eq false
+      end
+    end
+
+    context "when the param is a number" do
+      let(:kernel_param) { "32M" }
+
+      it "reports presence of the param" do
+        expect(is_crashkernel_set).to eq true
+      end
+
+      it "schedules the writing on the param" do
+        expect(crashkernel_needed).to eq true
+      end
+
+      it "does not find several ranges" do
+        expect(several_ranges).to eq false
+      end
+
+      it "correctly reads the size" do
+        expect(allocated_mem).to eq "32"
+      end
+    end
+
+    context "when the param is a range" do
+      let(:kernel_param) { "64M-:32M" }
+
+      it "reports presence of the param" do
+        expect(is_crashkernel_set).to eq true
+      end
+
+      it "schedules the writing on the param" do
+        expect(crashkernel_needed).to eq true
+      end
+
+      it "does not find several ranges" do
+        expect(several_ranges).to eq false
+      end
+
+      it "correctly reads the size" do
+        expect(allocated_mem).to eq "32"
+      end
+    end
+
+    context "when the param includes several ranges" do
+      let(:kernel_param) { "-200M:32M,200M-:64M" }
+
+      it "reports presence of the param" do
+        expect(is_crashkernel_set).to eq true
+      end
+
+      it "schedules the writing on the param" do
+        expect(crashkernel_needed).to eq true
+      end
+
+      it "finds several ranges" do
+        expect(several_ranges).to eq true
+      end
+
+      it "correctly reads the size of the last range" do
+        expect(allocated_mem).to eq "32"
+      end
+    end
+  end
+
+  describe ".WriteKdumpBootParameter" do
+    context "during autoinstallation" do
+      before do
+        allow(Yast::Mode).to receive(:autoinst).and_return true
+        allow(Yast::Mode).to receive(:update).and_return false
+        allow(Yast::Mode).to receive(:normal).and_return false
+        Yast::Kdump.Import(profile)
+      end
+
+      context "if kdump is requested and a value for crashkernel is supplied" do
+        let(:profile) { {"add_crash_kernel" => true, "crash_kernel" => "the_value"} }
+
+        it "writes the crashkernel value to the bootloader and enables the service" do
+          expect(Yast::Bootloader)
+            .to receive(:modify_kernel_params)
+            .with(:common, :xen_guest, :recovery, {"crashkernel" => "the_value"})
```

Comment:
only think that can work is with block and checking last parameter, but it looks like overkill for me. And also for crashkernel all of them are important as it mean it set value for common, xen_guest and also recovery kernel, but do not set value for xen_hostm so it have some semantic value.